### ISSUE TITLE
Groups split

### ISSUE THREAD

**Issue by darobin** [w3c/charter-html#14]
Just some thinking about a split somewhat different from Adrian's (but in keeping with his basic ideas). The proposal includes splitting HTML across the groups (as well as inside the groups).
- **Web Infrastructural Kernel**. This needs to be the most boring WG ever created. It includes DOM, the eventing system and event loop, WebIDL, URL, anything related to `public-script-coord`, parsing HTML, MIME sniffing, Window, the core algorithms in the HTML specification (parsing attribute values and the such), the HTML DOM (History, Location...), Streams, the Browsing the Web section.
- **Markup & Interactivity**. HTML overview specification, HTML best practices (anything that is now document conformance and the such), various specifications for attributes and elements of HTML, Web Components, Rendering HTML (default CSS and friends).
- **Platform Services**. Storage and network APIs, Service Workers, Clipboard, Selection, Editing, Packaging, Manifest, IME, UI Events.
- **Web Media**. EME, MSE, transcript, track binding, HTMLMediaElement, audio and video elements, possibly canvas.

**Comment by rubys** [comment]
@darobin: I like your thinking.  Can I ask your thoughts on what a "boring" URL spec would look like?

**Comment by darobin** [comment]
@rubys: Don't take this the wrong way, but I think it's already boring enough to make the cut :) The background is this: if we call something "Web Core" (or anything like that that implies some sort of super-powerful fundamental group) we're setting ourselves up for people to try to push their pet ideas into the core (just like people instinctively don't like their specs being an extension spec of HTML, even though it makes no difference in terms of conformance).

My inspiration there is `public-script-coord`. Very boring name, lots of discussion about extremely important topics that are pretty much the opposite of glamorous. I think that's what we want for that group.

**Comment by rubys** [comment]
> @rubys: Don't take this the wrong way, but I think it's already boring enough to make the cut :)

I'm not taking it the wrong way, but I will disagree with you.  :-)  And propose an alternative.

The current URL spec is very controversial in the IETF.  It contains best practices and conformance criteria (things you are proposing splitting out of the core HTML spec).  It contains things that no browser implements.  Overall, browser vendors do not consider it a priority to address differences between what the spec says and what they implement.  This is even true in cases where the spec and all other browser vendors agree.

If you take out all of this, what I see left is a definition of application/x-www-form-urlencoded and the URL APIUtils API (a.k.a., the URL object).  That and perhaps something akin to the paragraphs that are currently in the HTML spec's reference section as an introduction.

I believe that these parts meet your criteria for being important and boring.

**Comment by LJWatson** [comment]
@darobin

I like this structure. The grouping is simple, and the remit of each group seems reasonably clear.

Presumably accessibility would fall under the remit of the Markup & Interactivity group?

**Comment by plehegar** [comment]
@rubys, I don't think you're disagreeing with @darobin. The work in "Web Infrastructural Kernel" will be important, but that's not a reason for everyone to be part of that Group.

**Comment by plehegar** [comment]
@darobin, canvas in Web Media seems out of place. Web Media should be about timed based media imho and canvas doesn't seem out of place there. I wonder if canvas can be handed over to SVG folks and/or if they'd be interested.

**Comment by rubys** [comment]
@plehegar: I'm not disagreeing with @darobin, but I don't think I've made my point clearly.  Permit me to try again.

A while back, I noted that the split between WebApps and HTML sucked and suggested that they be merged.

@adrianba didn't disagree with me, but he took it further.  Once you put all the specs into the same pile, he noted that you could then split them out into an order that made more sense.

@darobin didn't disagree with @adrianba, but noted that you didn't have to presume that the specs themselves had to remain as is: in particular the HTML spec has boring parts and contentious parts.

And this brings us full circle.  I fully agree with @darobin, but would like to take his proposal further.  He put URL in one WG, but I believe that it too has boring parts and contentious parts.  @darobin and I would certainly agree on where the boring parts go.  I'm not sure yet where the contentious parts should go, but the answer is certainly: "any place but **Web Infrastructural Kernel**".  For now, I would suggest that the contentious parts go nowhere as addressing these issues does not appear to be a browser priority.

Clearer?

**Comment by darobin** [comment]
I think that splitting URL into boring "how browsers process URLs", "best practices for URLs", "URL API" (or whatever similar lines make sense) is worth exploring. It's an idea that's come up before in order to make the various parties happ(y|ier).

**Comment by plehegar** [comment]
regarding the split of URL, I can't imagine splitting the URL work into different Groups and how it would fit with the split proposed by Robin. I have no concern with splitting the spec within the same Group.

**Comment by rubys** [comment]
> I can't imagine splitting the URL work into different Groups and how it would fit with the split proposed by Robin

I can.  :-)  Specifically:
- "window.URL" goes into the same WG as "window.document"
- "best practices for URLs" goes into the same WG as "HTML best practices".

**Comment by plehegar** [comment]
Here is an other alternative split:
- **Web Infrastructural Kernel**. Includes HTML, DOM, Web Components, WebIDL, URL, anything related to `public-script-coord`, MIME sniffing
- **UI Core**.  Clipboard, Selection, Editing, IME, UI Events.
- **Platform Services**. Storage and network APIs (Web Storage, IndexedDB, Fetch, XHR, Beacon), Service Workers, Packaging, Manifest, Streams
- **Web Timed Media**. EME, MSE, transcript, track binding, HTMLMediaElement Core extensions, WebVTT, etc.

2DContext would be given to SVG Group.

Pros of this proposed split: doesn't separate the work of the DOM from its elements, doesn't make us wonder how to split the HTML spec.

**Comment by LJWatson** [comment]
@plehegar 
The UI Core group makes sense in some respects, but it feels a bit awkward to me. Things like HTML, Web Components etc. are also relevant to the UI, but they're grouped elsewhere.

**Comment by adrianba** [comment]
@LjWatson, the items proposed for "UI Core" revolve around a common theme of input/output interaction. Do you think the scope makes sense but has the wrong name or that the scope is troublesome too?

**Comment by darobin** [comment]
Reflecting a bit more on the HTML split, I think we don't have to make an agreement on the split as a pre-requirement for establishing the groups, but we should charter the groups in such a manner that they are allowed to take over parts of the HTML specification without rechartering.

**Comment by richschwer** [comment]
Regarding Canvas 2D Context and moving to SVG, does 2D Context have enough industry uptake?  I don't see a lot of work being done on it.

**Comment by marcoscaceres** [comment]
@richschwer see: http://caniuse.com/#search=canvas (it's very well supported)

**Comment by LJWatson** [comment]
@adrianba

“the items proposed for "UI Core" revolve around a common theme of input/output interaction. Do you think the scope makes sense but has the wrong name or that the scope is troublesome too?”

The scope doesn’t feel right, but on further reflection it might be due to a different interpretation of “UI”.

If it’s user-interaction, the grouping suggested by @plehegar makes sense. If user-interface in the wider sense, then the split doesn’t work as well -  because HTML and Web Components (at the least) feel as though they should fall under that definition.

**Comment by LJWatson** [comment]
@darobin

How might that work? Wouldn’t the charters need to be reasonably clear about which deliverables were within a group’s remit?

**Comment by richschwer** [comment]
@marcoscaceres WebGL is an entirely different API than Canvas 2D. Is it being suggested that this be taken up by the SVG WG as well? 
Also, I also see a LOT of gaps in IE support for Canvas 2D features. It appears that IE11 is not strategic for Microsoft so it would be good to see what is supported in Edge.

**Comment by marcoscaceres** [comment]
> @marcoscaceres WebGL is an entirely different API than Canvas 2D. Is it being suggested that this be taken up by the SVG WG as well? 

No. WebGL is handled by a different consortium (https://www.khronos.org/).

> Also, I also see a LOT of gaps in IE support for Canvas 2D features. It appears that IE11 is not strategic for Microsoft so it would be good to see what is supported in Edge.  

Edge support is also shown in the tables (just under IE11). Also, it's hyperbole to say there are a lot of gaps. IE11's support is on part with Gecko's (67%). That's not bad and I would expect it to get better over time (same with Gecko).

**Comment by plehegar** [comment]
@LjWatson re UI Core, yes, it would a User Interaction group.

**Comment by plehegar** [comment]
@darobin I believe writing charters in such a way that we can shuffle things around would be possible to a certain extend. If we define the scope properly, we can be more liberable on the set of deliverables. The AC has become less picky on scope/deliverables that it used to be. The best example is the 2014 CSS charter, that basically plenty of freedom to the CSS Working Group: "This list is not exclusive: The WG may also create new specifications, within its scope."
 http://www.w3.org/Style/2014/css-charter

**Comment by LJWatson** [comment]
@plehegar

Thanks for clarifying. That removes my concern about the grouping.

**Comment by darobin** [comment]
@LjWatson @plehegar I was also thinking about the HTML-WebApps conduit which allowed for the transfer of deliverables from the former to the latter. It's been in place for years, I wouldn't expect an outcry against it.

@richschwer @marcoscaceres It's worth noting that @sideshowbarker and I were talking to someone on the Khronos board recently and they were very interested in setting up collaboration with graphics work at W3C. I certainly wouldn't expect SVG to take over WebGL, but if given 2DContext it would be a logical place to bring in Khronos as well.

**Comment by LJWatson** [comment]
@darobin @plehegar

Good point. That conduit works, so no immediately obvious reason why it shouldn’t work in different circumstances.

**Comment by richschwer** [comment]
@darobin  OK.

**Comment by plehegar** [comment]
Some still see the idea of a split as a "W3C overhead". My next step on this front is to make sure we capture the rational (#18) and the motivations to justify a split of groups properly.

**Comment by adrianba** [comment]
> The AC has become less picky on scope/deliverables that it used to be.
> @plehegar, at least from our perspective that is more that we're resigned to having failed at getting better charters. :)

I do think is important for groups is to have a good strong definition of scope, especially as we create a new structure. One thing to consider for deliverables outside the current HTML/WebApps set is to explicitly declare them in scope for the new group so people do the IPR review while not necessarily transferring the work immediately. This way, when the participants find an appropriate time to move, the group doesn't have to recharter. This is especially important if the number of these features is more than one or two.

**Comment by adrianba** [comment]
> Some still see the idea of a split as a "W3C overhead".

This is an important point. "WebApps is working so why are we doing extra process to 'fix' something that isn't broken?"

The reason I proposed something like this is because I think HTML _is_ broken and I was hoping we could recruit the help of WebApps to fix it. With HTML5 at Recommendation, now seems like a good time to reflect and see if we can make improvements. Potentially short term pain for longer term gain.

**Comment by wayneca** [comment]
> The AC has become less picky on scope/deliverables that it used to be. 
> The best example is the 2014 CSS charter, that basically plenty of freedom 
> to the CSS Working Group: "This list is not exclusive: The WG may also create 
> new specifications, within its scope."

One reason clear scope and deliverables are important is that for WGs participating creates licensing obligations for all specs produced, not just one's own contributions like in Community Groups. So, it's important for people who worry about patents to be able to tell what they'd likely be licensing.  For CSS, it's pretty clear what they do.  For Web Apps, on the other hand, its "technologies related to developing client-side applications on the Web" and "server-side APIs for support of client-side functionality will be defined as needed".  That narrows it down :).  But, they have a very clear list of deliverables and the two combined make it so someone can tell what kind of patents may be relevant.  It has to be either the scope is narrow and well defined or the deliverables are.

**Comment by wayneca** [comment]
> One thing to consider for deliverables outside the current HTML/WebApps 
> set is to explicitly declare them in scope for the new group so people do 
> the IPR review while not necessarily transferring the work immediately

Starting the new WGs and saying what will be in them, and then transferring later is a really good idea. One of the concerns about this is stopping work to transfer it.  Getting the WGs started ahead of time can give time for handling permissions to join while the work continues where it is coming from. It would also help if the new WG Charter could say what it will take from what WG.  Then if Members are already in the donor WGs, approval for the new WG(s) can be simpler, because that stuff is already approved.

**Comment by AFBarstow** [comment]
Microsoft asserts in (1) a problem is "we don’t know that anybody actually wants to work on HTML5 maintenance in the next 12 months" and proposes the solution is to double the number of groups including splitting WebApps into three groups. This seems like a rather draconian (and punitive?) measure for WebApps, which IMHO is making forward progress in several areas.

A reorganization proposal without a clear strategic basis is simply Make Work à la  (2) (s/company/consortium/) and although it would likely generate lots of (needless) "activity", what we actually need is real and measurable "forward motion". Such proposals should include a compelling benefit/cost analysis, as well as agreed metrics to measure and evaluate their success/failure.

Additionally, with more and more spec work being done via Github, and thus facilitating more participation from the various stakeholders in the HTML ecosystem, group boundaries continue to be increasingly irrelevant (and perhaps even harmful), thus creating more group structures exacerbates the coordination functions and member/staff overhead we presumably want to reduce.

It would be substantially less disruptive, and address the stated problem, to simply add HTML to WebApps (and if/when the activity level becomes problematic within WebApps' current work mode and structure, we deal with that problem).

BTW, if the active participants of the HTMLWG agree the B/C ratio strongly favors forking the media related specs into a separate WG, then by all means just do it.

(1) http://github.adrianba.net/webstandards/HTML-innovation-vs-maintenance#to-recharter-a-html-working-group-or-not
(2)
http://lowres.cartoonstock.com/business-commerce-motto-mission_statement-look_busy-work_ethics-motivation-mfln7133_low.jpg

**Comment by adrianba** [comment]
> Microsoft asserts in (1) a problem is "we don’t know that anybody actually wants to work on HTML5 maintenance in the next 12 months"

This is misquoting and substantially different than what I wrote. In fact, I said that we do need to do maintenance work. In particular, we will need to do this for found interop issues.

Separately I said that there were several ideas for _new work_ in HTML but that these didn't appear to be a priority and that there is little evidence that people want to actually work on these in the next 12 months. If we rely on this new work to justify a HTML WG, which was a large part of the original HTML WG proposal, then we will be disappointed.

> This seems like a rather draconian (and punitive?) measure for WebApps.

Was it draconian and punitive to move DOM and Template into HTML. What about when Web Messaging, WebSockets, Web Workers moved from HTML to WebApps? I don't believe the proposal is to break WebApps into 3 groups and certainly not as a punishment.

> A reorganization proposal without a clear strategic basis is simply Make Work

I thought I had laid out a strategic basis but happy to address specific questions.

> It would be substantially less disruptive, and address the stated problem, to simply add HTML to WebApps (and if/when the activity level becomes problematic within WebApps' current work mode and structure, we deal with that problem).

This was my starting point and something I've proposed to you in the past. I received lots of feedback that this wasn't a good way forward so I tried to come up with something more structured. I'm certainly not wedded to any particular refactoring.

**Comment by rubys** [comment]
Here's the full paragraph in question:

> My worry is that there isn’t enough work to do just doing maintenance on HTML 5.0 and while we have some ideas for new work we could do we don’t know that anybody actually wants to work on them in the next 12 months. If we relaunch the working group but then do no work will we accomplish our goal? I think we only get once chance to do this so we need to have confidence that it will work.

Just so that it is clear, I'll speak in the first person about my beliefs.

I do believe it is likely that if you polled a large group of people a substantial majority would indicate that the W3C should maintain the specs that are published as Recommendations.  This would include HTML5.

I believe that the place to start is with the triaging of bugs.  That's not currently happening.  I don't believe that doing this without substantial direct vendor participation is a useful exercise.

As to the last sentence in the paragraph I quoted (namely the one containing "get one chance"), I don't believe that to be true.  Or more precisely, to the extent that it is true, that one chance happened long ago for many.

For that reason, I believe that "substantially less disruptive" is a non-goal.

There are some specs that are making forward progess in the HTML WG.  There are some specs that are making forward progress in the WebApps WG.  There also are some specs that are languishing in each.  I don't believe that changing the title of the associated working group would change things.  As a specific example, URL is languishing, and moving it to another WG wouldn't change matters.

From my perspective, we are experiencing a contraction in participation.  I am not sold on the idea that an expansion of the number of WGs is an appropriate response to that fact.  Collapsing the number of Working Groups with a commitment to spin off new groups that can demonstrate momentum and critical mass (Media being an obvious example) is a superior approach than trying to decide now where people might participate in the future.

Please don't take any of this to suggest that the structures proposed by @adrianba, @plehegar, or @darobin aren't plausible or defensible or would make things substantially worse.  I just don't believe that they address the root problem.

I'd prefer something more bottom up.  What specs are people interested in actively contributing to?  I'd like to suggest that those people get together, F2F if possible.  These people should start with bugs and test results and meet periodically; leading up to the point where an "Intent to Migrate" (or whatever that form is called at the moment) is produced.

I would eagerly participate in such -- whether is was in a role as co-chair, editor, tester, or humble secretary.  Before I do so, I would like to see first hand actual evidence that there is a critical mass of participants, and that critical mass includes browser vendors.  To put another way, I personally would not like to repeat the experience I had with URL.

**Comment by plehegar** [comment]
Here is the set of options I believe we have on the table. At the end of the day, it comes down to: who is going to step up and go proper maintenance of HTML if we're really serious about it.
- Option 1: HTML Working Group

Pros: Keep current repartition

Cons: Maintenance only?, separation of DOM/Web Components, will they come?
- Option 2: Merge HTML and WebApps Working Groups

Pros: DOM goes back to WebApps

Cons: Super Super Group
- Option 3: Split HTML Working Group

HTML, DOM, URL goes to WebApps. Media goes to Media WG.

Pros: Compromise between a full HTML/WebApps split and keeping HTML 
as-is. No disruption to WebApps. Gradual approach: could split further 
in the future if needed.

Cons: Still too much for WebApps to chew?
- Option 4: Split HTML/WebApps Working Group

Some combination around Infrastructure, Storage/Networking, Media, UI 
groups (ie 4 groups)

Pros: Rearrange the work in more meaningful modules

Cons: Highly disruptive for WebApps, Unnecessary W3C overhead?
- Option 5: Let the HTML Working Group lapse

Pros: it will happened naturally by the end of June

Cons: lack of leadership, inappropriate

**Comment by rubys** [comment]
> At the end of the day, it comes down to: who is going to step up and go proper maintenance of HTML if we're really serious about it.

+1

I suggest that picking options 1 through 4 without identifying who those people are isn't either leadership or appropriate.  It is folly.

I've heard second hand that those people exist.  I've also heard second hand from people who have talked F2F with **some of those very same people** who are hearing a completely different story.

Once those people self identify (preferably publicly, after all the results will be public), we should listen to how they would prefer to be organized.

**Comment by marcoscaceres** [comment]
> Once those people self identify (preferably publicly, after all the results will be public), we should listen to how they would prefer to be organized.

I agree what @rubys is suggesting. I have not heard of anyone willing to do the maintenance long-term, either - so I'm interested to see who they are. 

FWIW, I'm personally comfortable with option 5 (as I think the WHATWG community will do a reasonable job at maintaining the spec - as they have a strong vested interest to do so). On the W3C side, we have more important specs that need support and resource focus right now.

**Comment by wayneca** [comment]
If no one wants to work on HTML5 maintenance, moving it somewhere else where there are other interesting things, and putting it in its own GitHub repository where everyone is likely to ignore it, won't fix the problem. 

There seem to be a couple of variants in what's option 1.  There could be something called the HTML5 Maintenance WG aimed at compatibility corrections.  Anything new is done elsewhere like in WebApps.  The HTML Maintenance WG could use the public-review-announce list to ask Browser vendors to review specific changes in the Spec so they don't have to wade through all the discussions in the WG.  They can also use the process for revising the spec and I'd think the Browser folks would pay attention when it got to that level of actual changes (and not have to track lots of discussions that don't get to that point).  And anything significant could move out to WebApps as an extension spec. (and the Maintenance group could move extensions back into the HTML spec - or modularize it -- but not create major new things - they'd get those from the extensions done elsewhere).

**Comment by marcoscaceres** [comment]
> If no one wants to work on HTML5 maintenance, moving it somewhere else where there are other interesting things, and putting it in its own GitHub repository where everyone is likely to ignore it, won't fix the problem. 

Evidently, this is not the situation today: You can see that the maintenance on the WHATWG side is still happening fairly actively. https://html5.org/tools/web-apps-tracker

Equally so for picture: https://github.com/ResponsiveImagesCG/picture-element/commits/gh-pages/source

> There seem to be a couple of variants in what's option 1. There could be something called the HTML5 Maintenance WG aimed at compatibility corrections...

This doesn't seem workable to me, because it assumes that there is only 1 HTML spec (and throws us back into the dysfunctional situation we have today). This is not the case, as we know that the WHATWG is the upstream for the W3C HTML spec. Hence, it's better to allow changes to happen in the upstream (WHATWG) version. That is to say, if there is stuff to be fixed that is part of HTML, it should be filed as a bug on the WHATWG spec and let them fix it (unless it's one of the small bits of stuff where the W3C and WHATWG spec disagree) - and then those changes eventually get down-streamed to the HTML version. Eventually, a new HTML snapshot should be published a by the W3C (maybe once a year, depending on the changes).

**Comment by paulbrucecotton** [comment]
> •Option 3: Split HTML Working Group

In my role as HTML WG co-chair and in my role of leading the Media Task Force, I prefer this option.

> •Option 4: Split HTML/WebApps Working Group

I can live with this option.  Some of the spin off proposed here could be done as a second step from Option 3.

> •Option 5: Let the HTML Working Group lapse

I am STRONGLY against this option since I believe it would send the wrong message about the future of HTML/Web Platform at W3C and could be a PR nightmare for the W3C.

/paulc

PS: I hope the other WebApps and HTML WG co-chairs will clearly state their preferences on this Issue.

**Comment by marcoscaceres** [comment]
@chaals + @AFBarstow, see above. 

> I am STRONGLY against this option since I believe it would send the wrong message about the future of HTML/Web Platform at W3C and could be a PR nightmare for the W3C.

I don't see the negatives - or why it would be a "PR Nightmare"? The open nature of the way HTML was, and continues to be, developed naturally lends itself to option 5. The WHATWG was the original birthplace of HTML5, and ~99% of the text comes from it,  hence it makes sense that they continue to maintain it. They (well, mostly Hixie) have done a good job all these years at maintaining it - and generally I don't think anyone would disagree with that (even if views differ on some aspects of the spec). The real value that the W3C brings is the licensing (by virtue of it continuing to be a "RECOMMENDATION", and the licensing commitments that come with that - now and in the future). 

The PR story here is that the W3C needs to be ready to commit resources to the new stuff that keeps the web competitive: literally, **the next evolution of HTML and the Web platform**, which is what the ICG effort is all about, and what WebApps, WebAppSec, and WebPerf, and other groups at the W3C are already doing - these are great success stories and things that the W3C can be extremely proud of and that no one can fault them on. Freeing resources from HTMLWG means more resources can be assigned to a range of exciting and innovative new platform features (including new HTML features!). The W3C is not letting go of HTML, it's putting it's maintenance in trusted hands - and then releasing snapshots that reflect bug fixes to the spec. In parallel, it's focusing on standardizing the future of the platform. 

What would be a PR nightmare is that the W3C says "we are going to maintain it" and then we don't - or we do a bad job at it, and no one ends up actually referencing or implementing from the spec.

**Comment by LJWatson** [comment]
If option 2 or option 3 was chosen, it seems likely that work would be divided amongst TFs within the parent group. If so, the con for either option is essentially the same - the ability to effectively manage a "super super group".

The key difference between options 2 and 3 (as outlined by @plehegar), is whether there should be a Media WG or not. This in turn begs the question - what differentiates media from any of the other areas of work that would otherwise be subsumed into the "super super group"?

FWIW, I agree with PaulC that letting HTML lapse would send out the wrong message to the web community.

**Comment by marcoscaceres** [comment]
> FWIW, I agree with PaulC that letting HTML lapse would send out the wrong message to the web community.

@LjWatson, @paulbrucecotton, can you explain why? I want to understand your perspective more.

**Comment by paulbrucecotton** [comment]
> I want to understand your perspective more.

Again with my co-chair hat on - When the HTML WG co-chairs and Team presented our proposed After HTML5 plan to the W3C Advisory Board earlier this year, we received a clear message that it was not appropriate to delegate or assign the maintenance of a W3C Recommendation outside the organization.

In addition the W3C has collected together many new members and web related initiatives over the last 3 years that are all related to HTML5 and it simply would send the wrong message to those members and efforts to suggest that the W3C is not going to maintain the HTML5 Recommendation and/or work on HTML5 interoperability problems/issues.

I understand that you and others may disagree with the relative importance of these points but I am responding since you specifically asked me to explain my perspective.

/paulc

**Comment by marcoscaceres** [comment]
@paulbrucecotton thank you! I was not involved or aware of previous discussions, so this is very useful information to know.  

> we received a clear message that it was not appropriate to delegate or assign the maintenance of a W3C Recommendation outside the organization.

Ok, but this would have hinged on the assumption that there would be people willing to maintain the spec. I guess we need to wait till someone puts up their hand to do that? 

> In addition the W3C has collected together many new members and web related initiatives over the last 3 years that are all related to HTML5 and it simply would send the wrong message to those members and efforts to suggest that the W3C is not going to maintain the HTML5 Recommendation and/or work on HTML5 interoperability problems/issues.

But this assumes that there are significant issues to work on (are there?). What are some of those issues? What can the W3C HTML WG do that can't be handled by the WHATWG?  

> I understand that you and others may disagree with the relative importance of these points but I am responding since you specifically asked me to explain my perspective.

Certainly not downplaying or disagreeing with anything here. I'm just trying to get some of the background and a sense of where things are at with respect to HTML.

**Comment by LJWatson** [comment]
@marcoscaceres
"can you explain why? I want to understand your perspective more."

With my AC hat on, TPG is deeply invested in HTML at the W3C. We wouldn't want to see responsibility for this W3C standard to be delegated elsewhere.

With my me/engineer hat on, it seems logical to work on HTML under the W3C umbrella, because it's easier to collaborate with groups working on other specs - like ARIA for example.

**Comment by paulbrucecotton** [comment]
> But this assumes that there are significant issues to work on (are there?). 

There are currently 333 open HTML5 bugs [1] in the HTML WG HTML5  Bugzilla Component that need to triaged and then a decision made on how they should be handled.  Some will be editorial, some will be new features, some will indicate interop problems and other will probably point to possible spec changes.

/paulc

[1] http://tinyurl.com/ovrfah2

**Comment by stevefaulkner** [comment]
On 3 June 2015 at 20:39, Marcos Caceres notifications@github.com wrote:

> and ~99% of the text comes from it

some exaggeration, it is fair to say a large part (about 85%)

There are significant differences in normative and informative author
guidance for use of HTML,

The ARIA section of the whatwg spec has been stale for years now and
diverges from implementations.

I would be extremely concerned about leaving maintenance of the definition
of the HTML language to the WHATWG.

## 

Regards

SteveF
Current Standards Work @W3C
http://www.paciellogroup.com/blog/2015/03/current-standards-work-at-w3c/

**Comment by LJWatson** [comment]
@paulbrucecotton

With apologies for temporarily hijacking this discussion... we had some success with a bug triage team within the HTML A11y TF. I chaired it for a time, so if I can help with the outstanding HTML bug triage, let me know? I suspect time/charter may not be on our side, but just in case...

**Comment by rubys** [comment]
I agree that we should talk about messages.  That includes messages that the W3C should be looking for.  And messages that the W3C should be sending.  And for purposes of this discussion, I don't think it matters whether it is an explicit message or doing something that is likely to be noticed.  And similarly, I don't think it matters whether the root cause was inaction (allowing a charter to expire) or explicit action.

I will agree with Paul as to the messages that we received from the AB.  It is a shame that this message isn't public, but I will confirm that that is the message we received.  I will further state that I don't think it likely that we would get a different answer if the question were posed again.  And I do think that this input is something that the W3C should be listening to.

Marcos and I clearly agree that another source we should be listening to is people who are willing to do the work of triaging bugs and working to get implementations changed.  That being said, I disagree with Marcos that the record shows that there is active maintenance on the WHATWG spec.  Two substantive changes in the last 120 days is not active maintenance, not with the current bug backlogs and the size of the spec.  I'll also add another link for discussion: https://whatwg.org/issues/data.html?period=12

I will assert we shouldn't listen exclusively to only one of these sources, but rather that we should look at both.  And I think the latter is sending out the clear message that "A vocabulary and associated APIs for HTML and XHTML" is not currently a priority.  Very much related: the same is true for URLs.  Instead you see active discussions in topics like media, web components, service workers, streams and elsewhere.

My conclusion remains that listening to the AB and ignoring the very clear (albeit implicit) message that the actions (or more precisely the inactions) of those who the AB feels should be working the bug backlog is neither leadership nor appropriate.

I feel that the message that we should be sending is:

1) things are cyclic.  At the present time interests have moved from vocabularies to components.  This is the message being sent.  And the W3C is listening.  Message received.

2) the very instant that people once again start showing up to work on vocabularies, there is a welcome home for this work to occur at the W3C.  The model for that work won't be mailing lists and decision policies.  The model for that work will be based on github work flows and WebApps's workmode.

If the very act of sending out this message causes people to show up (like it has done with Léonie: much appreciated!), then so much the better!  I can't imagine anybody having a problem with any of options 1 through 4 if we have people actually showing up, doing the work, demonstrating results, and indicating a clear preference for any or all of these options.

Lacking such a critical mass, choosing options 1 through 4 at this time is not leadership and not appropriate.  It is wilfully ignoring the data that we have.

**Comment by marcoscaceres** [comment]
>  Two substantive changes in the last 120 days is not active maintenance.

Indeed. I stand corrected.

**Comment by paulbrucecotton** [comment]
@LjWatson 

> I chaired it for a time, so if I can help with the outstanding HTML bug triage,

60 of the outstanding 333 bugs contain the "a11y" keyword.  See http://tinyurl.com/q78atzs 

We should definitely consider deciding if/when to categorize these "a11y" bugs.

/paulc

**Comment by richschwer** [comment]
+1

**Comment by richschwer** [comment]
That was a +1 to Steve's comments.

**Comment by chaals** [comment]
I'll give myself an action to start bug triage in the accessibility task force and get those 60 chased up.

**Comment by darobin** [comment]
@marcoscaceres Right, as things stand today, we (the community "we", not any specific gang) appear to have an HTML maintenance problem. We need to tend to that.

No one is going to step up to the monster. The size can be daunting. It feels like a political bomb waiting to bite your head off at any time. It also has a some level of idiosyncrasy in the way in which it is put together that steepens the learning curve.

Beyond the rigmarole of group splitting, which has its usefulness but is less important, I think the only way we can get continued maintenance is by splitting HTML into enough pieces that people can feel confident just picking one up. The fact that the platform is largely a tangled bag doesn't mean it needs to be reflected as such at the document level.

We could discuss an ideal split of the document until the cows die of boredom. Instead I'm going to generate a split automatically, propose it, and the automatic part means that we can work over the details and tweak the split easily. I hope to have that next week.

In any case, this should not hold up organising the groups. We could follow the option of a minimal rearrangement there, and refine it by transferring specs between groups, splitting off units, etc.

It might, however, hold up part of the bug triage (but not all of it). If we're going to split the document, it's useful to know where the issues go. (Which could also be the occasion to move from Bugzilla to GH, if that's possible.)

**Comment by marcoscaceres** [comment]
> Beyond the rigmarole of group splitting, which has its usefulness but is less important, I think the only way we can get continued maintenance is by splitting HTML into enough pieces that people can feel confident just picking one up. The fact that the platform is largely a tangled bag doesn't mean it needs to be reflected as such at the document level.

It would be worth trying that, I guess. I'm worried about the intertwined-nature of the spec making that a complicated task. That is, even if you end up with features specs, the underlying mechanics still need to be understood. However, if this is mostly about maintenance, it could be feasible if we can find people that actually understand the work - and are willing to do the maintenance on the W3C side, rather than the WHATWG side.    

> We could discuss an ideal split of the document until the cows die of boredom. Instead I'm going to generate a split automatically, propose it, and the automatic part means that we can work over the details and tweak the split easily. I hope to have that next week.

Ok, let's see what comes out.

> In any case, this should not hold up organising the groups. We could follow the option of a minimal rearrangement there, and refine it by transferring specs between groups, splitting off units, etc.

Ok, but this again comes back to @rubys point about finding people to do the work. Or if groups (including competent editors) have agreed to take on the work. 

> It might, however, hold up part of the bug triage (but not all of it). If we're going to split the document, it's useful to know where the issues go. (Which could also be the occasion to move from Bugzilla to GH, if that's possible.)

I think web midi moved the issues over from bugzilla to GH, IIRC. So, at least there is some code around if you want to save yourself some time.

**Comment by darobin** [comment]
On 04/06/2015 17:35 , Marcos Caceres wrote:

> It would be worth trying that, I guess. I'm worried about the
> intertwined-nature of the spec making that a complicated task. That is,
> even if you end up with features specs, the underlying mechanics still
> need to be understood. However, if this is mostly about maintenance, it
> could be feasible if we can find people that actually understand the
> work - and are willing to do the maintenance on the W3C side, rather
> than the WHATWG side.

I agree completely. But I think it might actually make things clearer: 
some splinters will clearly be maintenance-only (even possibly nothing 
done at all because nothing is needed), whereas parts under active 
maintenance may be more clearly identified.

It's not trivial, of course, but I reckon it's worth a shot.

> I think web midi moved the issues over from bugzilla to GH, IIRC. So, at
> least there is some code around if you want to save yourself some time.

Were they the ones who did that _after_ setting up notifications for 
transferred comments and who DoSed the W3C email server then? ;-)

## 

Robin Berjon - http://berjon.com/ - @robinberjon

**Comment by marcoscaceres** [comment]
> Were they the ones who did that _after_ setting up notifications for transferred comments and who DoSed the W3C email server then? ;-)

heh, I don't think so - but can't be sure. At least, I don't have any memory of that happening.

**Comment by LJWatson** [comment]
@marcoscaceres
"It would be worth trying that, I guess. I'm worried about the intertwined-nature of the spec making that a complicated task. That is, even if you end up with features specs, the underlying mechanics still need to be understood. However, if this is mostly about maintenance, it could be feasible if we can find people that actually understand the work - and are willing to do the maintenance on the W3C side, rather than the WHATWG side."

Do we have a clear understanding of what the maintenance task will involve? Finding the right person/people will be easier if we know what needs to be done.

That said, if we don't know what the maintenance task looks like, there's a lot to be said for making a decision and iterating our approach as we go. My hunch is that we have both the ability and the capacity to do this - but the trouble with all of this, is that we won't really know until we try.

**Comment by marcoscaceres** [comment]
@LjWatson 

> Do we have a clear understanding of what the maintenance task will involve? Finding the right person/people will be easier if we know what needs to be done.

I guess we just need to take a look at the [bugs](http://tinyurl.com/ovrfah2) and see how many of those could be knocked off and by whom. 

> That said, if we don't know what the maintenance task looks like, there's a lot to be said for making a decision and iterating our approach as we go. My hunch is that we have both the ability and the capacity to do this - but the trouble with all of this, is that we won't really know until we try.

Maybe we could pick five bugs and see how we go?

**Comment by rubys** [comment]
> I guess we just need to take a look at the <a href="http://tinyurl.com/ovrfah2">bugs</a> and see how many of those could be knocked off and by whom. 

+1

**Comment by LJWatson** [comment]
@marcoscaceres
"Maybe we could pick five bugs and see how we go?"

Good idea. If we could gather a small group of people that collectively had a broad understanding of the different aspects of the spec (as it stands), we should be able to get a sense of the size/shape of the maintenance task.

Tangentially related... in response to the comment from @paulbrucecotton yesterday, we're putting the a11y TF bug triage team back together. @stevefaulkner @cptvitamin and I plan to tackle the heap of outstanding a11y bugs.

**Comment by cwilso** [comment]
> Were they the ones who did that after setting up notifications for transferred comments and who DoSed the W3C email server then? ;-)

It was Web Audio, actually (we never had a Web MIDI bugzilla, IIRC).  And yeah, I vaguely remember that might have happened.  :)

**Comment by marcoscaceres** [comment]
@cwilso, remember, Jussi did the import? For example: 
https://github.com/WebAudio/web-midi-api/issues/1

**Comment by cwilso** [comment]
I'm 100% on board with Robin's plan above.  To be clear, I think getting people to show up willing to work on the problems is the key.  I think getting lots of people to show up to one "room" is frequently an anti-pattern.  I'm not a fan of PLH's options 1,2 or 5.  I _suspect_ that Adrian's breakdown is probably close to what I'd expect to see; but even that is not really the point.  I think we need to look for the natural lines in the platform and figure it out from there.

Sam, I'm not sure what "message you received from the AB" that you're not comfortable sharing.  Mail me separately, and I'll get permission to share that publicly.

**Comment by cwilso** [comment]
@marcoscaceres oh, god, yeah.  Forgot we DID have a few issues before that.

**Comment by dontcallmedom** [comment]
re issue migration from bugzilla to github, I did it for the WebRTC WG (inspired by what the audio wg had done), and learned in the process how to do it even better than what I had managed (thanks to a contact at github); if and when this becomes relevant, ping me and I'll dig up the info.

**Comment by rubys** [comment]
> Sam, I'm not sure what "message you received from the AB" that you're not comfortable sharing. Mail me separately, and I'll get permission to share that publicly.

Um, I am comfortable sharing.    What I presume you are referring to is my statement "I will agree with Paul as to the messages that we received from the AB. It is a shame that this message isn't public, but I will confirm that that is the message we received".

If you are willing, my preference is that the AB should state directly what Paul relayed, and I confirmed as my understanding: 

https://github.com/w3c/charter-html/issues/14#issuecomment-108606731

It is always better when people get to hear these types of things first hand.  That being said, in this case, I think it is unlikely that both Paul and I heard this wrong, but you never know.

**Comment by rubys** [comment]
> I think getting people to show up willing to work on the problems is the key.
> I think we need to look for the natural lines in the platform and figure it out from there.

+1 to these two statements from @cwilso 

My suggestion for the way forward is to 
- do option 1 for six months (i.e., get a charter extension for the HTML WG)
- if people do show up during that time, take option 5 off the table
- take remaining viable options in reverse order

The current split between HTML and WebApps is of historical interest only, and super-super groups are problematic for a number of reasons.  Take time, figure out the right split, along the way verify that people are actually willing to show up, and then execute.

**Comment by adrianba** [comment]
> @cwilso wrote I'm not a fan of PLH's options 1,2 or 5. I suspect that Adrian's breakdown is probably close to what I'd expect to see; but even that is not really the point. I think we need to look for the natural lines in the platform and figure it out from there.

I'm not wedded to any particular split. As I've written about before, I don't think option 1 will work, at least in the short term. I don't like option 5 because we do have a desire to do maintenance work on HTML5 for issues we see affecting actual web sites today and causing problems where sites work in some browsers and not others (and often not in ours hence the desire to resolve them :smile:).

At TPAC last year, I spoke to many people to propose what amounts to option 2 above (merge HTML and WebApps). I heard lots of pushback and much of it was about the size and scope of the resulting "mega group". This is why I have more recently proposed a more granular refactoring. I can live with option 2, 3, or 4. Option 4 is essentially what I've been suggesting. I've heard the feedback that the split might be premature though. I see option 2 as do the merge now and then consider splitting off in future if necessary and not if not.

I like option 3 because the Media TF in the HTML WG has been discussing various things that wouldn't currently fall in scope for HTML or WebApps. Media is also complex area as far as IP goes. When we created the media TF in HTML one of the motivating reasons was that there were organisations who wanted to participate but only in the media related activities. I worry that if we ask their legal departments to do an IP review of the whole scope of HTML and WebApps then we might lose them. In addition, there are several Timed Media related activities from outside HTML/WebApps that might be good to bring together to ensure we have a cohesive approach.

I will take an action to try to draft an outline of what a charter for such a Media WG might look like so we have something more concrete to consider.

I'm also working with @frankolivier at Microsoft to identify some of the HTML5 issues we've run into that are examples of the kind of HTML maintenance we want to do. I will try to share these later today.

One of the things I'd like to figure out is how we differentiate maintenance (which in my mind is something like bug fixing of issues in HTML 5.0) from adding net new features to HTML (which I'd prefer to be in new specs and ideally initially incubated in the CG). There is a grey area in the middle where we're adding new functionality but maybe it is such a small extension that writing a new spec if hard to justify. I'm not sure how to propose to handle that. Perhaps Robin's experiment with the split will make it easier.

**Comment by cwilso** [comment]
If you mean

> ...we received a clear message that it was not appropriate to delegate or assign the maintenance of a W3C Recommendation outside the organization.
> Then I can ask the AB to directly state that if it will help, but that is most definitely my recollection as well.  Yes, of course, we need people to show up as well.

**Comment by cwilso** [comment]
> do option 1 for six months (i.e., get a charter extension for the HTML WG)

I'm not strictly opposed to that, but I think that's mostly buying time in order to refine how to restructure.

> if people do show up during that time, take option 5 off the table

I don't see that as a cause and effect - unless you really do mean ONLY the option 5 translation of "stop caring about HTML at W3C".  Regardless of whether people show up or not, I think the split in HTML today (i.e., not a split) is not a healthy one, and I'm somewhat ambivalent about whether there's a WG named "HTML WG" in the future.  I do think just dropping driving HTML like a hot rock is crazy, too.

> The current split between HTML and WebApps is of historical interest only, and super-super groups are problematic for a number of reasons. Take time, figure out the right split, along the way verify that people are actually willing to show up, and then execute.

+1.

**Comment by sideshowbarker** [comment]
> There are currently 333 open HTML5 bugs [1] in the HTML WG HTML5 Bugzilla Component that need to triaged and then a decision made on how they should be handled. Some will be editorial, some will be new features, some will indicate interop problems and other will probably point to possible spec changes.
> 
> /paulc
> 
> [1] http://tinyurl.com/ovrfah2

I will spend some time on triaging the non-a11y bugs in that list.

**Comment by rubys** [comment]
> >   do option 1 for six months (i.e., get a charter extension for the HTML WG)
> 
> I'm not strictly opposed to that, but I think that's mostly buying time in order to refine how to restructure.

Exactly.  Nothing more.  Nothing less.  And note, once that time has expired, this is my least favorite option.  And judging by other responses, it seems to be the least likely end game. 

> >  if people do show up during that time, take option 5 off the table
> 
> I don't see that as a cause and effect - unless you really do mean ONLY the option 5 translation of "stop caring about HTML at W3C". Regardless of whether people show up or not, I think the split in HTML today (i.e., not a split) is not a healthy one, and I'm somewhat ambivalent about whether there's a WG named "HTML WG" in the future. I do think just dropping driving HTML like a hot rock is crazy, too.

My opinion is that if people don't show up, the question as to how the W3C chooses to split the work is moot.

**Comment by rubys** [comment]
People who have indicated an interest in help triaging the backlog of HTML bugs to date: @darobin, @LjWatson @chaals, @sideshowbarker ; if I missed anybody, please speak up.

**Comment by marcoscaceres** [comment]
:raising_hand:

**Comment by stevefaulkner** [comment]
me too, I have already made a small start on bugs

## 

Regards

SteveF
Current Standards Work @W3C
http://www.paciellogroup.com/blog/2015/03/current-standards-work-at-w3c/

**Comment by LJWatson** [comment]
@rubys
"People who have indicated an interest in help triaging the backlog of HTML bugs to date: @darobin, @LjWatson @chaals, @sideshowbarker ; if I missed anybody, please speak up."

@stevefaulkner and @cptvitamin also.

**Comment by LJWatson** [comment]
@adrianba
"One of the things I'd like to figure out is how we differentiate maintenance (which in my mind is something like bug fixing of issues in HTML 5.0) from adding net new features to HTML (which I'd prefer to be in new specs and ideally initially incubated in the CG). There is a grey area in the middle where we're adding new functionality but maybe it is such a small extension that writing a new spec if hard to justify. I'm not sure how to propose to handle that. Perhaps Robin's experiment with the split will make it easier."

This feels as though it could fit within the maintenance function (although "maintenance" may not be the best description for it). If we take "maintenance" to mean ongoing caretaking of the base spec, that could easily include the addition of new features not large enough to warrant exclusive development.

**Comment by adrianba** [comment]
> @LjWatson wrote: This feels as though it could fit within the maintenance function (although "maintenance" may not be the best description for it). If we take "maintenance" to mean ongoing caretaking of the base spec, that could easily include the addition of new features not large enough to warrant exclusive development.

Yes. It's just hard to draw the line sometimes. How much "new" do you need before it's no longer a bug fix and is now a new feature.

**Comment by hober** [comment]
> My opinion is that if people don't show up, the question as to how the W3C chooses to split the work is moot.

I think there's a chicken-and-egg problem here: without refactoring into more sensible groups, how do we sell people on the idea that the climate/culture has improved sufficiently that it might be worth coming back?

**Comment by rubys** [comment]
> I think there's a chicken-and-egg problem here: without refactoring into more sensible groups

@hober: I'd suggest that the problem is deeper than that... guessing at what refactoring people would consider sensible is less likely to be successful than gathering input on this matter.  Amongst employees of major browser vendors, only @adrianba has proposed a split.

**Comment by cwilso** [comment]
@rubys, Yes, only Adrian has proposed a split; I didn't realize you were wait for an explicit proposal from other vendors.  I've stated multiple times that I think Adrian's proposal is a good starting point.

FWIW, I think tightening the media relationship would be a good idea, both giving stronger management to EME/MSE and mapping a plan for the media bedrock implied in various WebRTC and Web Audio specs.  I haven't worked out a specific proposal there.

I think looking at the natural connections across HTML and the set of things in Web Apps, and plotting divisions based on shared work, would be a good way to start.

**Comment by timbray** [comment]
I blogged on this subject: https://www.tbray.org/ongoing/When/201x/2015/06/05/End-of-HTML

Tl;dr: HTML is good enough.  Lots of other pieces of the Web ecosystem aren’t, so I think letting the vocabulary lie fallow for a while would be a smart thing to do.

**Comment by darobin** [comment]
@timbray Right, because we have no interoperability issues with any part of HTML. Just like `contenteditable` just works! As does `accesskey`; and in general we don't have the problem that native apps are more accessible than HTML. And input modalities are a solved problem, in the same way that HTML forms rock the world and are totally devoid of issues.

I could keep on going...

**Comment by rubys** [comment]
@darobin: do we have evidence that the above list is a priority from either a user or vendor perspective?

@slightlyoff has two related tweets: guessing isn't progress (https://twitter.com/slightlylate/status/607300722741239809) and deck-chair movement (https://twitter.com/slightlylate/status/607301020805300225)

It would be nice if the collective response to the request to be more data-driven was something more than "don't understand; proceeding anyway".

A pause doesn't mean "never get back to it".

**Comment by adrianba** [comment]
I worked with @frankolivier to review bugs we would consider as higher priority maintenance work. Here are some examples:
- Syntax issue with meta refresh (see [bug 25407](https://www.w3.org/Bugs/Public/show_bug.cgi?id=25407)). This affects a large number of (especially mobile) sites and required a change in Microsoft Edge for web compat.
- Meta charset=utf8 is expected to be supported (i.e. without the dash so in addition to "utf-8").
- <button> elements inside <a> anchors are expected to work by sites

I also worked with @arronei to review the ~300 current bugs open against the HTML5 spec. I've taken Arron's analysis and converted it to HTML and [published it here](http://github.adrianba.net/webstandards/HTML5-bugs.htm). This is an attempt to triage the bugs into categories. This is a rough quick pass and the goal is to understand the relative magnitude of maintenance bugs vs. editorial bugs vs. new functionality. There are undoubtedly some mistakes but I think the table gives a good indication of the scale of the work in the current bug database.

**Comment by adrianba** [comment]
> I will take an action to try to draft an outline of what a charter for such a Media WG might look like so we have something more concrete to consider.

My early draft proposal is in the pull request #34. I have discussed this with @cwilso although he hasn't yet seen this text. I'm hoping for feedback about the Audio and WebRTC WG interactions. Also hoping @plehegar can provide feedback. And anyone else of course.

**Comment by chaals** [comment]
Archeology: 
As part of the AB, and without either speaking for them or publishing things people said on the understanding that they would be made available to the W3C members only, assigning maintenance outside the organisation strikes me as inappropriate because…
1) W3C and WHAT-WG HTML5 have diverged, in ways that matter to W3C, whose value proposition includes getting working accessibility and being seen to do so.
2) While WHAT-WG maintains some things others just get lost
3) if W3C produces a recommendation it has an obligation not to let that happen while the spec is still in use (the raison d'être of the WHATWG in the first place).
This is not to say that W3C should simply reject the idea, but that they have an absolute responsibility for maintenance of their specs which are actually relevant to their members - much as Ade has put together one possible triage…

**Comment by rubys** [comment]
@chaals: I'll note that number 2 above also applies to the W3C.  I'll further note that the "absolute responsibility" to which you refer falls not to the AB, but to the people who show up to do the work.

There is no question in my mind that people with an interest in accessibility will show up.  There is no question in my mind that the W3C staff and team will show up.  @adrianba (and therefore presumably Microsoft) have amply demonstrated that he/they are willing to do their part to make this work.

I'll continue to ask the uncomfortable questions: what about the other major browser vendors?  What happens if two or more wish to resume maintenance of what they refer to as the "upstream spec" elsewhere?

**Comment by darobin** [comment]
@adrianba I just wanted to say a big thank you to you, @frankolivier, and @arronei. Your bug table is awesome.

**Comment by jeffjaffe** [comment]
@rubys I believe Chaals "absolute responsibility" falls to the W3C, not to the AB (as you infered) - with which I concur.

**Comment by rubys** [comment]
@jeffjaffe we may have to agree to disagree on this.

The question is: if you knew with absolute certainly that a significant number of the necessary players were not going to show up, would it be (a) the responsible thing to do, or (b) absolutely irresponsible to propose a group charter including the task of maintenance of the HTML specification?

That being said, hopefully other browser vendors follow Microsoft's excellent lead and make this question moot.

Meanwhile, I heartily second @darobin's thank you.

**Comment by jeffjaffe** [comment]
@rubys If we have absolute responsibility, I would think that we are required to try to make it succeed.  I recognize your coaching that if we were to fail, we would need a responsible way to proceed in that case.

**Comment by rubys** [comment]
@jeffjaffe I would prefer that we avoid use of words like 'fail'.

As an example, I worked on the URL spec on the (now in hindsight, mistaken) impression that browsers felt it to be important to improve interoperability in this area.  Given the present level of interest exhibited, I think it is would be (and in the case of the WHATWG, is) irresponsible to publish that spec with the same level of endorsement than other (more widely implemented) specs have.

This was definitely a learning experience.  I would however only call it a failure if somebody identified something I should have or should be doing but am not.

That aside, I gather that you are (rather gently) suggesting that I make a responsible proposal.  Fair enough.

Given the current levels of participation, I would suggest that any charters that include HTML maintenance (I intentionally phrased it this way to make what I am proposing independent of any group split) be explicitly limited to what @adrianba, @frankolivier, and @arronei categorize as either editorial or P3.  I would also be willing to add accessibility changes that do not require browser implementations to change.

None of this should preclude P1 changes from happening in a community group.  Or for a rechartering to occur once it has been demonstrated that "wide review" means more than an opportunity to object was presented and nobody did so.

**Comment by chaals** [comment]
Some current thoughts:

Extend HTML-WG for 6 months.
Yes, this is just marking time. But there are a couple of pieces of work happening there. As Robin points out, there are also some things that need to be happening. And one of the urgent pieces should be to figure out how to do the work there is.

Make a media WG
This is the most obvious part of splitting things so that the people who are relevant to each piece focus on their are of interest and expertise, without having to winnow it out from lots of other stuff.

Propose a path forward **to the AC**
At some point we should formally ask if a strawman is good enough, instead of spending another six months refining it. A formal "vote" (OK, those don't exist, but an AC review is similar) finds things that a github discussion doesn't bring out.

Charter an HTML incubator CG
The fact that we have no idea how to do this, a couple of years after we started looking at how to put CGs into the process, suggests that there is work for the process wonks (i.e. those who get W3C members to agree on how W3C works). In the meantime, let's get working on the stuff we need.

@adrianba @frankolivier @arronel, Thank you. Awesome work.

**Comment by chaals** [comment]
@rubys if one browser vendor wants to work on the whatwg spec and not at W3C, I don't see the big issue. It just makes a bit more headache for everyone to keep things in synch, but in practical terms it is manageable - browser vendors are also effectively working on their own as they innovate and develop.

If most browser vendors don't want to work with W3C then there is a problem. It isn't clear to me that this is the case.

And while browser vendors are an important part of the ecosystem, they are not the holders of all power in practice, nor should they be in theory. They all have to implement things that are shipped by content producers from time to time.

Any viable HTML spec needs to have the engagement of more than just the four West Coast-based companies that some seem to assume are the be-all and end-all of the web.

**Comment by rubys** [comment]
@chaals: I agree fully with https://github.com/w3c/charter-html/issues/14#issuecomment-111074081

As to your comment that follows, not so much.

> if one browser vendor wants to work on the whatwg spec and not at W3C, I don't see the big issue. It just makes a bit more headache for everyone to keep things in synch, but in practical terms it is manageable - browser vendors are also effectively working on their own as they innovate and develop.

It has been more than one, and it has been a big issue.  @darobin's spork should help, but  is based on the controversial premise that initial bug resolution is delegated.  Having the HTML WG "resolve" bugs only to have a different resolution appear later in the WHATWG spec is a bit more than a headache.  As a practical matter, the editorial team had a full time job doing merges leaving little time for new development.

Furthermore, this is not the problem we have.  Browser vendors aren't working elsewhere on the HTML vocabulary, they mostly are working on other things.

> If most browser vendors don't want to work with W3C then there is a problem. It isn't clear to me that this is the case.

Only one browser vendor to date has demonstrated that they are willing to work through the bug backlog.  It is a problem.

> And while browser vendors are an important part of the ecosystem, they are not the holders of all power in practice, nor should they be in theory. They all have to implement things that are shipped by content producers from time to time.

Strawman.  Nobody is saying that the HTML WG should consist exclusively of browser vendors.  As an example, I am personally on record arguing against private meetings limited only to browser vendors and the W3C team that determine priorities, as I feel that such meetings can neither be honestly referred to as "side" meetings, nor do they match the openness values that the W3C espouse.

> Any viable HTML spec needs to have the engagement of more than just the four West Coast-based companies that some seem to assume are the be-all and end-all of the web.

Nobody is saying any different.  What has been said is that unless the HTML spec does have the engagement of a majority of browser vendors, changes should be limited to areas that don't require browser vendors to make changes.

**Comment by AFBarstow** [comment]
In https://github.com/w3c/charter-html/issues/14#issuecomment-111074081 Chaals said:

[[
Charter an HTML incubator CG

The fact that we have no idea how to do this, a couple of years after we started looking at how to put CGs into the process, suggests that there is work for the process wonks (i.e. those who get W3C members to agree on how W3C works). In the meantime, let's get working on the stuff we need.
]]

I think part of the problem with this CG proposal is that it includes formalities of a Working Group, has a broad scope and modus operandi that is closer to an Interest Group, yet wants to be an incubator for new ideas, akin to a Community Group. Additionally, the planned use of dedicated staff resource(s) is contrary to the general expectation for CGs.

It seems like it would be simpler and potentially more encompassing to create an IG that shepherds/coordinates/faciliates/... the various proposals and that the proposals are fleshed out in CGs such that each CG can create their own rules, policies, work mode, etc. The IG should create guidelines and best practices that cover topics such as "importance of (tracking) contribution provenance", "importance of consensus", "how to get a CG spec on the Recommendation track", "importance of broad buy-in", etc. and advocate their usage. The IG would also be responsible for helping these CGs transition their work to the Recommendation track (f.ex. getting a spec to an existing WG or starting a new WG).

**Comment by adrianba** [comment]
> and that the proposals are fleshed out in CGs such that each CG can create their own rules, policies, work mode, etc.

The proposal for an incubator CG was specifically to remove this friction from getting started. There's nothing preventing people creating many CGs today but all too often I am approached with a proposal to "just work in GitHub" because figuring out rules, policies, and work mode is too much trouble.

At this point it seems that some people are supportive of the idea for an incubator CG and others are sceptical. I think the current discussion has been helpful but, since anybody can create a CG, I'm hopeful we can just try it soon and see if it works and adjust as we go if not. Or stop if it really doesn't work.

**Comment by LJWatson** [comment]
@rubys 
"Only one browser vendor to date has demonstrated that they are willing to work through the bug backlog. It is a problem."

By my reckoning two browser vendors (Microsoft and Yandex) have offered to invest time in bug triage, and a third (Google) has actively contributed to this discussion.

Do we know whether anyone has reached out to Apple, Mozilla or opera to ask them to consider helping out? If not, it might be a bit hasty to claim lack of interest.

**Comment by jeffjaffe** [comment]
@LjWatson +1.  The way to build momentum to work on this problem is to bootstrap; a little interest; driving more interest - until we get to critical mass.  We can't expect people to show up at the party if we have not infused the party with renewed energy.

**Comment by rubys** [comment]
@LJWatson: I think you missed one: @marcoscaceres works for Mozilla.  Other than that, I just think that you and I are applying different metrics to the term "actually demonstrated".

**Comment by cwilso** [comment]
And, in fact, both Google and Apple volunteered to participate in a constrained effort to solve the URL spec issue as well.    The way to get these problems solved are to break them down into constrained efforts, get resources assigned to work on them, and then solve those constrained problems.

+1 to Adrian's comments on the CG; I would not want to sign up to having an IG in addition.  We need fewer layers of process, not more.

**Comment by dwsinger** [comment]
> On Jun 11, 2015, at 12:40 , Chris Wilson notifications@github.com wrote:
> 
> And, in fact, both Google and Apple volunteered to participate in a constrained effort to solve the URL spec issue as well. The way to get these problems solved are to break them down into constrained efforts, get resources assigned to work on them, and then solve those constrained problems.
> 
> +1 to Adrian's comments on the CG; I would not want to sign up to having an IG in addition. We need fewer layers of process, not more.

agreed on both, and catching up.  (Apologies, I somehow failed to sign up for notifications from this conversation, and didn’t notice it going on.  Mea culpa, need Robin’s dashboard.)

random thoughts:
- It makes sense to re-factor HTML, URL and WebApps. But it can be a goal to get to a more reasonable structure, it doesn’t need to be done in one swell Foop! and it surely should not be disruptive.
- as part of the URL spec., the whole question of how one defines URLs forms for a new scheme is another area that needs attention but is a whole different skill-set
- my guess is that the AC needs WG charters to have a clear scope, but that scope can be defined/delineated in multiple ways: by a list of deliverables, or (as in the case of CSS) a well-defined technical area; I expect there are other ways, too
- I like some of the re-structuring ideas here; I also like working on what matters
- On bugs: my company doesn’t fix immediately every bug I report (or anyone else, for that matter). Similarly, that there are bugs in HTML that we don’t feel urgently need fixing — or we need experience to know how to fix — should come as no surprise.  The mere existence of un-fixed bugs is not, IMHO, an issue.
- Is our maintenance procedure too heavyweight? have we actually written down what it takes to get a fix into say the HTML spec.? I think the perception may be that you need to report it, identify the fix, do the editing, do the work to get a revised-Rec through the process, and so on.  I’ve talked about ‘living errata’ pages before, and so on.  Maybe we need a process that adopts a fix if X% of the WG affirmatively say ‘that’s right’ and no-one objects, for example? (This is very strawman-y).
- people turn up because they believe people will turn up; it is a confidence question.  we need to instill that confidence.  then they stay when they find the right people and infrastructure to get things done (or to have a say in what gets done).
- yes, we care about HTML, both bugs and new features: just not all of them all of the time

David Singer
Manager, Software Standards, Apple Inc.

**Comment by plehegar** [comment]
After several back-and-forth, the proposal is to merge the HTML and Web Applications Working Groups together, while creating a Media Working Group. Several reasons for doing so:
- Documents like DOM and URL done in the HTMLWG came from WebApps originally. Web Components, seen as the next HTML big frontier, is currently in WebApps. The overlap between HTML and WebApps work has been there for several years (as proven by some of the current wordings in the HTML and WebApps charters).
- Splitting HTML and WebApps groups would be too disruptive at this time and we probably wouldn't get the split right
- HTML needs to be properly maintained and have a path for evolution, even if some may not see the need to evolve it substantially at the moment
- Merging HTML and WebApps does keep the split option open while we're looking at splitting the HTML specification (see also https://lists.w3.org/Archives/Public/public-html/2015Jun/0009.html). We'd like to revisit this question next year. W3C is moving its workmode towards using github and the number of working groups may not make a difference in this move anyway (besides adding overhead). 
- the work of the media task force has been done in a separate task force and we need to properly ensure that the core media layer of Web is properly defined for the purpose of the Extensible Web. See a draft charter for this at http://w3c.github.io/charter-html/timed-media-wg.html .

The proposal will be to have those two Groups operating under the W3C Document and Software license for their deliverables:
 http://www.w3.org/Consortium/Legal/2015/copyright-software-and-document.html

**Comment by plehegar** [comment]
By "those two Groups", I do mean the merged HTML/WebApps one and the media one.

**Comment by plehegar** [comment]
We now have a draft charter for the merged HTML/WebApps group:
 http://w3c.github.io/charter-html/group-charter.html
Ideas for a new name (ie other than WebApps) are welcome.

**Comment by silviapfeiffer** [comment]
Maybe web client interfaces WG can work?

**Comment by LJWatson** [comment]
@plehegar
"Ideas for a new name (ie other than WebApps) are welcome."

Throwing some ideas out (off the top of my head)...
Web Interfaces, Web Interface Technologies, Web Infrastructure, Web Infrastructures, Web Semantics and Structures, Web Semantics and Interfaces.

Failing that, I suppose "Web Stuff" would be out of the question!

**Comment by wayneca** [comment]
Web Platform WG?

**Comment by cwilso** [comment]
"Honey Pot WG".  :P

**Comment by adactio** [comment]
[Web Thangs Working Group](https://adactio.com/journal/6246)?

**Comment by stevefaulkner** [comment]
Web Features Working Group?

**Comment by hober** [comment]
So far, I like "Web Platform WG" the best.

**Comment by darobin** [comment]
I'm with @hober.

**Comment by marcoscaceres** [comment]
/me tries to come up with a React Native joke and fails.... so yeah, Web Platform WG.

**Comment by plehegar** [comment]
I'm not a fan of Web Platform because of the honey pot risk but we may not be able to avoid in any case. So https://github.com/w3c/charter-html/commit/dc98f862a1242c056d82f72d1fc4463188677926 for now

**Comment by silviapfeiffer** [comment]
Web Core WG?

**Comment by chaals** [comment]
I'm seriously not a fan of bikesheds (I was thinking Bikeshed Any Thing So HTML Is Timely, but couldn't be bothered reverse-engineering an acronymic for crazy).

Web Applications has been around as a name for a decade or so, and has apparently seemed clear enough to us, to WHATWG, and the universe. What's broken that we need to fix?

**Comment by chaals** [comment]
Although anything that has WC or WFM as an acronym works too.

**Comment by cwilso** [comment]
Web Technology Foundations Working Group.  :P

"Web Core" is the least-best of all options, imo.

**Comment by AFBarstow** [comment]
On 6/19/15 6:50 PM, chaals wrote:

> I'm seriously not a fan of bikesheds (I was thinking Bikeshed Any 
> Thing So HTML Is Timely, but couldn't be bothered reverse-engineering 
> an acronymic for crazy).

So, I guess the following are out: "Web Hypertext Applications and Tags 
WG" and "Browser Stuff WG" ;-)

> Web Applications has been around as a name for a decade or so, and has 
> apparently seemed clear enough to us, to WHATWG, and the universe. 
> What's broken that we need to fix?

I agree with this sentiment, and have a relatively strong preference to 
include  "Web Applications" in the group's title, although appending 
HTML would be fine too since the proposal is to literally create a "Web 
Applications and HTML WG".

**Comment by adactio** [comment]
[The web is not a platform](https://adactio.com/journal/6692).

[The web is not a platform](https://www.youtube.com/watch?v=r38al1w-h4k).

[The web is not a platform](https://medium.com/@jackysee/the-web-is-not-a-platform-9ac5ae8d3b77).

**Comment by dwsinger** [comment]
> On Jun 20, 2015, at 5:19 , Arthur Barstow notifications@github.com wrote:
> 
> On 6/19/15 6:50 PM, chaals wrote:
> 
> > I'm seriously not a fan of bikesheds (I was thinking Bikeshed Any 
> > Thing So HTML Is Timely, but couldn't be bothered reverse-engineering 
> > an acronymic for crazy).
> 
> So, I guess the following are out: "Web Hypertext Applications and Tags 
> WG" and "Browser Stuff WG" ;-)

Web Applications and Platform (WAP)?

:-)

David Singer
Manager, Software Standards, Apple Inc.

**Comment by dwsinger** [comment]
It’s quite commonly thought of as such, and indeed in many ways it supplied a platform for all sorts of things.  So, perhaps you could explain as well as assert?

> On Jun 20, 2015, at 20:20 , Jeremy Keith notifications@github.com wrote:
> 
> The web is not a platform.

David Singer
Manager, Software Standards, Apple Inc.

**Comment by adactio** [comment]
David Singer wrote:

> It’s quite commonly thought of as such, and indeed in many ways it supplied a platform for all sorts of things.  So, perhaps you could explain as well as assert?

I did explain. The explanations can be found at the three hyperlinks I posted. Follow any (or all) of them for further explanation.

**Comment by marcoscaceres** [comment]
@adactio, you really are preaching to the converted. Your expectations of "platforms" don't appear the same as mine (and possibly others in this thread), which is understanding the web as a continuum of fault-tolerant technologies  (which appears to be the overarching theme of the articles you pointed to, as also your position). 

To call it "web spectrum" or "web continuum" or something other than "platform" would be academic at this point. The web **is** a platform that is capable of running on _multiple platforms_ - which is unlike other platforms. However, it still requires APIs, new protocols, and other specified behavior to make it interoperable and to continue to provide value to both users and developers - just like other software platforms. I don't think anyone here has lost sight of the Web's original goals of universal access, or the separation between content, structure, and markup, etc. At the same time, declarative aspects of the web are sometimes overly constraining (or just simply bad and wrong) - which is why a lot of the work being done to progressively enhance the Web has been happening over the last few years (e.g., Service Workers, Web Manifest, Web Components, etc.).

**Comment by darobin** [comment]
@adactio Of course the Web is a platform; it is both a nice surface on which people can make stuff stand and the common policy of people who believe in a worldwide shared information space. This has nothing to do with whatever axe one may wish to grind about Web applications or whatever else. Even 25 years ago it was a hypertext publishing platform.

**Comment by adactio** [comment]
@darobin Of course the Web is not a platform.

Weird—adding the words “of course” in there didn’t seem to have any noticeable effect.

Ah well, we’ll just agree to differ. But please do continue to characterise anyone who disagrees with your own viewpoint as “having an axe to grind”. That’s super-helpful.

**Comment by darobin** [comment]
@adactio Sure, I'm happy to just agree to disagree. It is, largely, terminology and in my honestly humble opinion where it isn't terminology I feel it is mostly a strawman. But FYI it's hard not to get an axe-to-grind impression from this:

![axe](https://cloud.githubusercontent.com/assets/38491/8283308/e38ae54c-18f6-11e5-8a74-4ed7ba47495c.png)

**Comment by dwsinger** [comment]
> On Jun 21, 2015, at 18:11 , Jeremy Keith notifications@github.com wrote:
> 
> David Singer wrote:
> 
> It’s quite commonly thought of as such, and indeed in many ways it supplied a platform for all sorts of things. So, perhaps you could explain as well as assert?
> 
> I did explain. The explanations can be found at the three hyperlinks I posted. Follow any (or all) of them for further explanation.

I am sorry, I should not have replied with jet-lag. I saw the repetition and underlining as both being some form of emphasis, whereas actually the underlining meant something else. My mistake, and apologies.

Nonetheless, I would disagree with what you say.

"And the web is not like that. The web is not binary, one or zero, on or off. It’s not a platform where you get one hundred per cent or zero per cent. It’s this continuum.
The web is not a platform. It’s a continuum.”

Who says a platform has to have hard ‘edges’ and be perfectly uniformly implemented everywhere? I think that the last pair of sentences presents a false dichotomy.

Also, in the tension between native and web, the web is attractive in very large part in that it offers a platform which is widely implemented on a whole variety of systems.  You want to be tightly integrated with, and optimized for, a specific system? — go native. You want broad deployability and system independence? — go to the web.

One _can_ deploy a user experience, an ‘application’ in the general sense, as a web experience, and have it work across a significant range of systems and devices, without excessive testing of what system you’re actually on (with care and luck, and if you stay away from the latest or experimental features, with no system-sensitivity). That is a platform, in many people’s understanding — something that they can stand on, that is essentially ‘opaque’ and offers an abstraction you don’t have to look under.

I don’t think that only one layer, the OS plus it services gets to claim the ‘platform’ name — indeed, as you say, Flash can be a platform for some purposes — and that the broadly-deployed web specifications DO offer an important platform for a whole class of services and experiences.

David Singer
Manager, Software Standards, Apple Inc.

**Comment by plehegar** [comment]
@adactio I'm failing to understand your proposal if you have one. What exactly would you like to see done here?

**Comment by chaals** [comment]
This issue seems to have been overtaken by events. Closing
